syntax = "proto3";

package chirp;

message PutRequest {
  bytes key = 1;
  bytes value = 2;
}

message PutReply {
  // Empty because success/failure is signaled via GRPC status.
}

message GetRequest {
  bytes key = 1;
}

message GetReply {
  bytes value = 1;
}

message DeleteRequest {
  bytes key = 1;
}

message DeleteReply {
  // Empty because success/failure is signaled via GRPC status.
}

service KeyValueStore {
  rpc put (PutRequest) returns (PutReply) {}
  rpc get (stream GetRequest) returns (stream GetReply) {}
  rpc deletekey (DeleteRequest) returns (DeleteReply) {}
}
